packages/aws-cdk-lib/aws-elasticloadbalancingv2/test/alb/target-group.test.ts

```diff
@@ -861,4 +861,155 @@ describe('tests', () => {
       });
     });
   });
+
+  describe('Lambda target multi_value_headers tests', () => {
+    test('Lambda target should have multi_value_headers.enabled set to true when enabled', () => {
+      // GIVEN
+      const app = new cdk.App();
+      const stack = new cdk.Stack(app, 'Stack');
+
+      // WHEN
+      new elbv2.ApplicationTargetGroup(stack, 'TG', {
+        targetType: elbv2.TargetType.LAMBDA,
+        multiValueHeadersEnabled: true,
+      });
+
+      // THEN
+      Template.fromStack(stack).hasResourceProperties('AWS::ElasticLoadBalancingV2::TargetGroup', {
+        TargetType: 'lambda',
+        TargetGroupAttributes: [
+          {
+            Key: 'lambda.multi_value_headers.enabled',
+            Value: 'true',
+          },
+        ],
+      });
+    });
+
+    test('lambda.multi_value_headers.enabled is not set when multiValueHeadersEnabled is false', () => {
+      // GIVEN
+      const app = new cdk.App();
+      const stack = new cdk.Stack(app, 'Stack');
+
+      // WHEN
+      new elbv2.ApplicationTargetGroup(stack, 'TG', {
+        targetType: elbv2.TargetType.LAMBDA,
+        multiValueHeadersEnabled: false,
+      });
+
+      // THEN
+      Template.fromStack(stack).hasResourceProperties('AWS::ElasticLoadBalancingV2::TargetGroup', {
+        TargetType: 'lambda',
+        TargetGroupAttributes: Match.absent(),
+      });
+    });
+
+    test('lambda.multi_value_headers.enabled is not set when multiValueHeadersEnabled is not specified', () => {
+      // GIVEN
+      const app = new cdk.App();
+      const stack = new cdk.Stack(app, 'Stack');
+
+      // WHEN
+      new elbv2.ApplicationTargetGroup(stack, 'TG', {
+        targetType: elbv2.TargetType.LAMBDA,
+      });
+
+      // THEN
+      Template.fromStack(stack).hasResourceProperties('AWS::ElasticLoadBalancingV2::TargetGroup', {
+        TargetType: 'lambda',
+        TargetGroupAttributes: Match.absent(),
+      });
+    });
```

Comment:
Could you please use `test.each`?

```suggestion
   test.each([false, undefined])('lambda.multi_value_headers.enabled is not set when multiValueHeadersEnabled is %s', (multiValueHeadersEnabled) => {
      // GIVEN
      const app = new cdk.App();
      const stack = new cdk.Stack(app, 'Stack');

      // WHEN
      new elbv2.ApplicationTargetGroup(stack, 'TG', {
        targetType: elbv2.TargetType.LAMBDA,
        multiValueHeadersEnabled,
      });

      // THEN
      Template.fromStack(stack).hasResourceProperties('AWS::ElasticLoadBalancingV2::TargetGroup', {
        TargetType: 'lambda',
        TargetGroupAttributes: Match.absent(),
      });
    });
```